api/v1alpha1/apparmorprofile_types.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type AppArmorProfileSpec struct {
+	Rules    string `json:"rules"`
```

Comment:
I think we could break this down into the different types of rules - similarly to what we have on the Seccomp CRD. This would allow us to validate profiles to quite a good degree.

Reply:
@pjbgf can you point me to a few examples?

Reply:
On Seccomp we define the different types here: https://github.com/kubernetes-sigs/security-profiles-operator/blob/b9715d8142237f0ea4b64ff9c569370369d6e2cd/api/seccompprofile/v1alpha1/seccompprofile_types.go#L25

That allows us to statically verify that a profile is valid, which thinking about it is a slightly easier job considering that seccomp profiles are generally represented in `json` already. Some tools (e.g. `aa-easyprof`) support json [representation](http://manpages.ubuntu.com/manpages/xenial/man8/aa-easyprof.8.html) of AppArmor profiles, but to my knowledge users tend to use the unstructured approach more often.

We don't need to take the same approach for the AppArmor policy content, but it is worth discussing how can we strike a balance between the user experience on using existing policies and being able to statically verify the policy at creation time.

Reply:
I think this would be worthwhile too